src/main/java/net/pms/dlna/ImageInfo.java

```diff
@@ -0,0 +1,621 @@
+					if (
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_XMIN) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_XMAX) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_YMIN) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_YMAX)
+					) {
+						width = ((PcxDirectory) directory).getInt(PcxDirectory.TAG_XMAX) - ((PcxDirectory) directory).getInt(PcxDirectory.TAG_XMIN) + 1;
+						height = ((PcxDirectory) directory).getInt(PcxDirectory.TAG_YMAX) - ((PcxDirectory) directory).getInt(PcxDirectory.TAG_YMIN) + 1;
+					}
+				} else if (directory instanceof PsdHeaderDirectory) {
+					if (
+						((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_IMAGE_WIDTH) &&
+						((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_IMAGE_HEIGHT)
+					) {
+						width = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_IMAGE_WIDTH);
+						height = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_IMAGE_HEIGHT);
+					}
+					if (((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_BITS_PER_CHANNEL)) {
+						bitDepth = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_BITS_PER_CHANNEL);
+					}
+					if (((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_COLOR_MODE)) {
+						/*
+					     * Bitmap = 0; Grayscale = 1; Indexed = 2; RGB = 3; CMYK = 4; Multichannel = 7; Duotone = 8; Lab = 9.
+						 */
+						switch (((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_COLOR_MODE)) {
+							case 1:
+								numComponents = 1;
+								colorSpaceType = ColorSpaceType.TYPE_GRAY;
+								break;
+							case 2:
+							case 3:
+								numComponents = 3;
+								colorSpaceType = ColorSpaceType.TYPE_RGB;
+								break;
+							case 4:
+								numComponents = 4;
+								colorSpaceType = ColorSpaceType.TYPE_CMYK;
+								break;
+							default:
+						}
+					}
+
+				} else if (directory instanceof WebpDirectory) {
+					if (
+						((WebpDirectory) directory).containsTag(WebpDirectory.TAG_IMAGE_WIDTH) &&
+						((WebpDirectory) directory).containsTag(WebpDirectory.TAG_IMAGE_HEIGHT)
+					) {
+						width = ((WebpDirectory) directory).getInt(WebpDirectory.TAG_IMAGE_WIDTH);
+						height = ((WebpDirectory) directory).getInt(WebpDirectory.TAG_IMAGE_HEIGHT);
+					}
+				}
+			}
+		}
+		if (throwOnParseFailure && (width < 0 || height < 0)) {
+			throw new MetadataException("Failed to parse image resolution from metadata");
+		}
+		this.width = width;
+		this.height = height;
+		this.format = format;
+		this.size = size;
+		this.bitDepth = bitDepth;
+		this.numComponents = numComponents;
+		this.colorSpace = null;
+		this.colorSpaceType = colorSpaceType;
+		this.metadata = metadata;
+		this.exifOrientation = orientation;
+		this.imageIOSupport = false;
+	}
+
+	/**
+	 * Tries to parse Exif orientation from the given metadata. If it fails in
+	 * in any way, {@code 0} (Normal) is returned.
```

Comment:
@Sami32 I first and foremost look at what you write, not the code you're commenting on, because I'm expecting your comment to point out what's wrong. To me, "any way, {@code 0} (Normal) is returned" looked perfectly ok. Had you simply written "in in", "in **in**" or better "in ~~in~~" I would have understood right away 😕 (I think!?)